src/django_components/component.py

```diff
@@ -54,37 +55,113 @@
 RENDERED_COMMENT_TEMPLATE = "<!-- _RENDERED {name} -->"
 
 
-class SimplifiedInterfaceMediaDefiningClass(MediaDefiningClass):
+class ComponentMeta(MediaDefiningClass):
     def __new__(mcs, name: str, bases: Tuple[Type, ...], attrs: Dict[str, Any]) -> Type:
         # NOTE: Skip template/media file resolution when then Component class ITSELF
         # is being created.
         if "__module__" in attrs and attrs["__module__"] == "django_components.component":
             return super().__new__(mcs, name, bases, attrs)
 
         if "Media" in attrs:
-            media: Component.Media = attrs["Media"]
+            media_data: Component.Media = attrs["Media"]
+            _normalize_media(media_data)
+            _normalize_media_filepaths(media_data)
```

Comment:
`_normalize_media_filepaths` what allows users to define file paths as str/bytes/PathLike/Callable